### PR TITLE
Refactore pipeline

### PR DESCRIPTION
**Pull request by sirh3e**

### CODE REVIEW

**Comment by jkotalik** [comment]
@sirh3e can you please combine all of your changes into a single PR instead of multiple PRs? Will be much easier for use to review.

**Comment by sirh3e** [comment]
@jkotalik sure but pls give one more day 😀

**Comment by davidfowl** [comment]
@jkotalik are you sure it'll be easier? These are all non functional changes...

**Comment by jkotalik** [comment]
Yeah I guess you are right.

File: src/Microsoft.Tye.Core/Pipeline.cs [hunk 1]
```diff
@@ -38,9 +38,9 @@ public static void Configure<TApplication>(Action<TApplication> callback)
             foreach (var (type, delegates) in callbacks)
             {
                 var method = pipeline.GetType().GetMethod("Register")!.MakeGenericMethod(type);
-                for (var i = 0; i < delegates.Count; i++)
+                foreach (var @delegate in delegates)
```
Comment (jkotalik) [inline src/Microsoft.Tye.Core/Pipeline.cs:41]: nit: `@delegate` to `del`

Comment (sirh3e) [reply src/Microsoft.Tye.Core/Pipeline.cs:41]: Why?

Comment (jkotalik) [reply src/Microsoft.Tye.Core/Pipeline.cs:41]: Personally not a fan of using `@` variables.

Comment (sirh3e) [reply src/Microsoft.Tye.Core/Pipeline.cs:41]: Delegate is much cleaner then del. Del could mean to much

**Pull request merged**